### PR TITLE
Fix compiling on M1 / arm64

### LINKED ISSUE #3: Gem fails to build native extensions on M1
```bash
% gem install fasttext -v '0.2.1' --source 'https://rubygems.org/'
Building native extensions. This could take a while...
ERROR:  Error installing fasttext:
	ERROR: Failed to build gem native extension.

    current directory: /Users/russ/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/fasttext-0.2.1/ext/fasttext
/Users/russ/.rbenv/versions/2.7.2/bin/ruby -I /Users/russ/.rbenv/versions/2.7.2/lib/ruby/2.7.0 -r ./siteconf20210629-98318-88rrv6.rb extconf.rb
checking for rice/rice.hpp in /Users/russ/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/rice-4.0.2/include... yes
checking for -lc++... yes
creating Makefile

current directory: /Users/russ/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/fasttext-0.2.1/ext/fasttext
make "DESTDIR=" clean

current directory: /Users/russ/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/fasttext-0.2.1/ext/fasttext
make "DESTDIR="
compiling ext.cpp
clang: error: the clang compiler does not support '-march=native'
make: *** [ext.o] Error 1

make failed, exit code 2

Gem files will remain installed in /Users/russ/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/fasttext-0.2.1 for inspection.
Results logged to /Users/russ/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/extensions/arm64-darwin-20/2.7.0/fasttext-0.2.1/gem_make.out
```

### PR DESCRIPTION
**Pull request by ukd1**
This fixes #3 for me, and should be the same on non-M1 / arm64 platforms.

### CODE REVIEW

**Comment by ukd1** [comment]
before;

```bash
russ@Russells-MBP fastText % bundle exec rake clean  
russ@Russells-MBP fastText % bundle exec rake compile
mkdir -p tmp/arm64-darwin20/ext/2.7.2
cd tmp/arm64-darwin20/ext/2.7.2
/Users/russ/.rbenv/versions/2.7.2/bin/ruby -I. ../../../../ext/fasttext/extconf.rb
checking for rice/rice.hpp in /Users/russ/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/rice-4.0.2/include... yes
checking for -lc++... yes
creating Makefile
cd -
cd tmp/arm64-darwin20/ext/2.7.2
/usr/bin/make
compiling ../../../../ext/fasttext/ext.cpp
clang: error: the clang compiler does not support '-march=native'
make: *** [ext.o] Error 1
rake aborted!
Command failed with status (2): [/usr/bin/make...]
/Users/russ/.rbenv/versions/2.7.2/bin/bundle:23:in `load'
/Users/russ/.rbenv/versions/2.7.2/bin/bundle:23:in `<main>'
Tasks: TOP => compile => compile:arm64-darwin20 => compile:ext:arm64-darwin20 => copy:ext:arm64-darwin20:2.7.2 => tmp/arm64-darwin20/ext/2.7.2/ext.bundle
(See full trace by running task with --trace)
```

after:

```bash
russ@Russells-MBP fastText % bundle exec rake compile
cd tmp/arm64-darwin20/ext/2.7.2
/Users/russ/.rbenv/versions/2.7.2/bin/ruby -I. ../../../../ext/fasttext/extconf.rb
checking for rice/rice.hpp in /Users/russ/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/rice-4.0.2/include... yes
checking for -lc++... yes
creating Makefile
cd -
cd tmp/arm64-darwin20/ext/2.7.2
/usr/bin/make
compiling ../../../../ext/fasttext/ext.cpp
compiling /Users/russ/Sites/fastText/vendor/fastText/src/dictionary.cc
compiling /Users/russ/Sites/fastText/vendor/fastText/src/main.cc
compiling /Users/russ/Sites/fastText/vendor/fastText/src/autotune.cc
compiling /Users/russ/Sites/fastText/vendor/fastText/src/fasttext.cc
compiling /Users/russ/Sites/fastText/vendor/fastText/src/utils.cc
compiling /Users/russ/Sites/fastText/vendor/fastText/src/model.cc
compiling /Users/russ/Sites/fastText/vendor/fastText/src/loss.cc
compiling /Users/russ/Sites/fastText/vendor/fastText/src/productquantizer.cc
compiling /Users/russ/Sites/fastText/vendor/fastText/src/args.cc
compiling /Users/russ/Sites/fastText/vendor/fastText/src/quantmatrix.cc
compiling /Users/russ/Sites/fastText/vendor/fastText/src/matrix.cc
compiling /Users/russ/Sites/fastText/vendor/fastText/src/meter.cc
compiling /Users/russ/Sites/fastText/vendor/fastText/src/vector.cc
compiling /Users/russ/Sites/fastText/vendor/fastText/src/densematrix.cc
linking shared-object fasttext/ext.bundle
cd -
mkdir -p tmp/arm64-darwin20/stage/lib/fasttext
install -c tmp/arm64-darwin20/ext/2.7.2/ext.bundle lib/fasttext/ext.bundle
cp tmp/arm64-darwin20/ext/2.7.2/ext.bundle tmp/arm64-darwin20/stage/lib/fasttext/ext.bundle
russ@Russells-MBP fastText % 
```

**Comment by ukd1** [comment]
+ tests on M1

```bash
russ@Russells-MBP fastText % bundle exec rake test   
Run options: --seed 58002

# Running:

Progress: 100.0% Trials:    7 Best score:  1.000000 ETA:   0h 0m 0s
Training again with best arguments
Read 0M words
Number of words:  14
Number of labels: 2
Progress: 100.0% words/sec/thread:    1020 lr:  0.000000 avg.loss:  0.694191 ETA:   0h 0m 0s
Progress: 100.0% Trials:    8 Best score:  1.000000 ETA:   0h 0m 0s
Training again with best arguments
Read 0M words
Number of words:  14
Number of labels: 2
Progress: 100.0% words/sec/thread:    1020 lr:  0.000000 avg.loss:  0.693408 ETA:   0h 0m 0s
Read 0M words
Number of words:  14
Number of labels: 2
Progress: 100.0% words/sec/thread:    1021 lr:  0.000000 avg.loss:  0.693427 ETA:   0h 0m 0s
Read 0M words
Number of words:  14
Number of labels: 2
Progress: 100.0% words/sec/thread: 1828073 lr:  0.000000 avg.loss:  0.693582 ETA:   0h 0m 0s
Read 0M words
Number of words:  14
Number of labels: 2
Progress: 100.0% words/sec/thread:     692 lr:  0.000000 avg.loss:  0.693986 ETA:   0h 0m 0s
Progress: 100.0% Trials:    7 Best score:  1.000000 ETA:   0h 0m 0s
Training again with best arguments
Read 0M words
Number of words:  14
Number of labels: 2
Progress: 100.0% words/sec/thread:    1015 lr:  0.000000 avg.loss:  0.693986 ETA:   0h 0m 0s
Read 0M words
Number of words:  5
Number of labels: 0
Progress: 100.0% words/sec/thread:     333 lr:  0.000000 avg.loss:       nan ETA:   0h 0m 0s
Read 0M words
Number of words:  5
Number of labels: 0
Progress: 100.0% words/sec/thread:     662 lr:  0.000000 avg.loss:       nan ETA:   0h 0m 0s
Read 0M words
Number of words:  5
Number of labels: 0
Progress: 100.0% words/sec/thread:     663 lr:  0.000000 avg.loss:       nan ETA:   0h 0m 0s
.

Fabulous run in 12.284384s, 0.8954 runs/s, 3.2562 assertions/s.

11 runs, 40 assertions, 0 failures, 0 errors, 1 skips

You have skipped tests. Run with --verbose for details.
russ@Russells-MBP fastText % 
```

File: ext/fasttext/extconf.rb [hunk 1]
```diff
@@ -1,7 +1,12 @@
 require "mkmf-rice"
 
 # -pthread and -O3 set by default
-$CXXFLAGS << " -std=c++17 $(optflags) -funroll-loops " << with_config("optflags", "-march=native")
```
Comment (ankane) [inline ext/fasttext/extconf.rb:4]: Let's only change the default on this line so `optflags` can still be passed. Also, for the arm64 check, let's use `RbConfig::CONFIG["host_cpu"] =~ /arm/i` so it's consistent with my other ML gems.

**Review by ankane — COMMENTED** [review]
Hey @ukd1, thanks for the PR! Left a comment inline.

File: ext/fasttext/extconf.rb (hunk 1 above)
Comment (ukd1) [reply ext/fasttext/extconf.rb:4]: By only the default, do you mean still pass ``optflags``, and only conditionally pass ``-march=native``?

Comment (ankane) [reply ext/fasttext/extconf.rb:4]: Yeah, I think we can do something like:

```ruby
default_optflags = RbConfig::CONFIG["host_os"] =~ /darwin/i && RbConfig::CONFIG["host_cpu"] =~ /arm/i ? "-march=armv8-a" : "-march=native"
... << with_config("optflags", default_optflags)
```

**Comment by ankane** [comment]
Fixed in #5. Thanks again for the PR!

**Pull request closed without merging**